wallet/src/Pos/Wallet/Web/Tracking/Sync.hs

```diff
@@ -387,52 +366,60 @@ trackingRollbackTxs (getEncInfo -> encInfo) allAddress getDiff getTs txs =
     foldl' rollbackTx mempty txs
   where
     rollbackTx :: CAccModifier -> (TxAux, TxUndo, BlockHeader ssc) -> CAccModifier
-    rollbackTx CAccModifier{..} (TxAux {..}, NE.toList -> undoL, blkHeader) = do
-        let hh = headerHash blkHeader
+    rollbackTx CAccModifier{..} (tx, undo, blkHeader) = do
+        let wh@(WithHash _ txId) = withHash (taTx tx)
+            hh = headerHash blkHeader
             hhs = repeat hh
-            mDiff = getDiff blkHeader
-            mTs = getTs blkHeader
-            tx@(UnsafeTx (NE.toList -> inps) (NE.toList -> outs) _) = taTx
-            !txid = hash taTx
-            undoL' = catMaybes undoL
-            resolvedInputs = zip inps undoL'
-            txOutgoings = map txOutAddress outs
-            txInputs = map (toaOut . snd) resolvedInputs
-
-            ownInputs = selectOwnAddresses encInfo (txOutAddress . toaOut) undoL'
-            ownOutputs = selectOwnAddresses encInfo txOutAddress $ outs
-            ownInputMetas = map snd ownInputs
-            ownOutputMetas = map snd ownOutputs
-            ownInputAddrs = map cwamId ownInputMetas
-            ownOutputAddrs = map cwamId ownOutputMetas
-
-            l = fromIntegral (length outs) :: Word32
-            ownTxIns = zip inps $ map fst ownInputs
```

Comment:
this line was wrong because if there were some `Nothing` in tx undo (for unknown addresses) we would have concatenated some elements from `ownInputs` with wrong `TxIn`